flink-connectors/flink-connector-kafka-base/src/test/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaProducerBaseTest.java

```diff
@@ -293,6 +293,61 @@ public void run() {
 		testHarness.close();
 	}
 
+	/**
+	 * This test is meant to assure that testAtLeastOnceProducer is valid by testing that if flushing is disabled,
+	 * the snapshot method does indeed finishes without waiting for pending records;
+	 * we set a timeout because the test will not finish if the logic is broken
+	 */
+	@Test(timeout=5000)
+	public void testDoesNotWaitForPendingRecordsIfFlushingDisabled() throws Throwable {
+		final OneShotLatch inputLatch = new OneShotLatch();
+
+		final DummyFlinkKafkaProducer<String> producer = new DummyFlinkKafkaProducer<>(
```

Comment:
Can't we simply give this instance a mock `KafkaProducer`? Then we could easily check via `verify` whether the `send` method has been called.